crypto/src/main/java/org/cryptimeleon/incentive/crypto/proof/spend/zkp/SpendDeductZkp.java

```diff
@@ -0,0 +1,35 @@
+package org.cryptimeleon.incentive.crypto.proof.spend.zkp;
+
+import org.cryptimeleon.incentive.crypto.model.IncentivePublicParameters;
+import org.cryptimeleon.incentive.crypto.model.PromotionParameters;
+import org.cryptimeleon.incentive.crypto.model.keys.provider.ProviderPublicKey;
+import org.cryptimeleon.incentive.crypto.proof.spend.tree.SpendDeductAndNode;
+import org.cryptimeleon.incentive.crypto.proof.spend.tree.SpendDeductTree;
+
+/**
+ * A wrapper class that helps building typical SpendDeduct ZKPs that all hvae a Metadata leaf and a protocol tree for
+ * conditions and updates each.
+ */
+public class SpendDeductZkp extends SpendDeductBooleanZkp {
+
+    /**
+     * Constructor.
+     *
+     * @param spendDeductConditionTree a SpendDeductTree that represents conditions on the old token which must be fulfilled
+     * @param spendDeductUpdateTree    a SpendDeductTree that represents conditions on the new token relative to the old token
+     * @param pp                       the public parameters
+     * @param promotionParameters      the promotion parameters
+     * @param providerPublicKey        the public key of the provider
+     */
+    public SpendDeductZkp(SpendDeductTree spendDeductConditionTree, SpendDeductTree spendDeductUpdateTree, IncentivePublicParameters pp, PromotionParameters promotionParameters, ProviderPublicKey providerPublicKey) {
+        super(
+                new SpendDeductAndNode(
+                        spendDeductConditionTree,
+                        spendDeductUpdateTree
+                ),
+                pp,
+                promotionParameters,
+                providerPublicKey
+        );
+    }
+}
```

Comment:
Hm. Hard to say. I feel like the ZK-generator class should just take both as input and "AND" them there, instead of having the API-user connect their two trees "manually"?! 
Is there a world where you'd ever do something like `conditionTree OR updateTree`? I'm thinking probably there is no real other choice than to `AND` the two types of trees 🤔. If that is the case, I'd definitely "AND" them at "ZK compile time" instead of here.

Reply:
Okay, I am not sure about what you mean with 'ZK compile time' though

Reply:
If the process is 

1. User instantiates the policy API tree stuff
2. User calls the method that assembles a ProofOfPartialKnowledge instance
3. User executes the protocol from step 2

Then "ZK compile time" is Step 2, i.e. converting the policy tree into a protocol.

Reply:
Okay, roger that.
One use-case/problem I stumbled upon is that an updateTree might be sufficient for describing a promotion (e.g. I reduce points by 5 and the new score is >=0). Furthermore, I think that condition+update OR condition2+update2 is a more more realistic use-case and would require removing this strict distinction between conditionTrees and updateTrees.
So to sum up, I would recommend changing the API from `f(conditionTree, updateTree)` to `f(spendDeductTree)`. The promotion api should make sure, that the resulting statements make sense.